assets/src/edit-story/app/story/useStoryReducer/reducers/addElements.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Internal dependencies
+ */
+import { isInsideRange } from './utils';
+
+/**
+ * Add elements to given page. If no page given, add elements to current page.
+ *
+ * If page index is outside bounds of available pages, nothing happens.
+ *
+ * Elements are expected to a be list of element objects with at least an id property.
+ * No validation is made on these objects. If elements aren't a list, nothing happens.
+ *
+ * Elements will be added to the front (end) of the list of elements on the current page.
+ *
+ * If elements are added to the current page, selection is set to be exactly the new elements.
+ * Otherwise, selection is unchanged.
+ *
+ * Current page is unchanged.
+ *
+ * @param {Object} state Current state
+ * @param {Object} payload Action payload
+ * @param {number} payload.pageIndex Page to insert elements on or null if current page should be used
+ * @param {Array.<Object>} payload.elements Elements to insert on the given page.
+ * @return {Object} New state
+ */
+function addElements( state, { pageIndex, elements } ) {
+	const indexToAddTo = pageIndex === null ? state.current : pageIndex;
```

Comment:
Nit, but maybe `state.currentPageIndex`?

Reply:
I considered it, but it's an internal variable only, so I went with only the three variables [`pages`, `current` and `selection`](https://github.com/ampproject/amp-wp/pull/3910/files#diff-018c60aaf786b41efe49a8ba88150c6fR12-R16).

They will never be used outside the scope of the reducer, so I went for brevity over specificity here.

I can revert though to more explicit variables, `currentPageId` and `selectedElementIds`.

Reply:
sg

Reply:
sg to the short names, i mean :)